activejob/lib/active_job/test_helper.rb

```diff
@@ -358,7 +358,13 @@ def assert_enqueued_with(job: nil, args: nil, at: nil, queue: nil)
       end
 
       matching_job = jobs.find do |enqueued_job|
-        serialized_args.all? { |key, value| value == enqueued_job[key] }
+        serialized_args.all? do |key, value|
```

Comment:
@rafaelfranca I've changed the technique to compare the arguments.
I don't really like it. It's too custom IMO.

Can you suggest some way to make this approach better?
Or else, do you think I should deserialize the arguments in `enqueued_job` and compare those with the un-serialized arguments passed to the test helper method? It will be a little bit cleaner that way. However, I am not sure how much of a performance hit it would be to deserialize all arguments for every `enqueued_job` in `jobs`(it might just be ok for tests, I don't know).

Reply:
I think it is fine to deserialize the arguments and compare them. We want to compare the arguments deserialized anyway.

We want to do:

```ruby
assert_enqueued_with(args: [Model.first])
```

Not `[Model.first.to_global_id]`

Reply:
I didn't completely understand the code example you posted but I'll go ahead and make the change to compare deserialized arguments and then we can discuss further.

Reply:
Got it :)